src/figuresClasses.ts

```diff
@@ -1,19 +1,98 @@
-export interface Figure {
+type Color = 'red'|'green'|'blue';
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
+}
 
+export interface Figure {
+  shape: Shape
+  color: Color
+  getArea():number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.triangle;
 
-}
+  constructor(
+    public color: 'red'|'green'|'blue',
+    public a: number,
+    public c: number,
+    public b: number,
+  ) {
+    this.color = color;
+
+    const longestSide = Math.max(a, b, c);
+    const otherSides = a + b + c - longestSide;
+    const notZero = a > 0 && b > 0 && c > 0;
 
-export class Circle {
+    if (longestSide >= otherSides || !notZero) {
+      throw new Error('your error message');
+    }
 
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+
+    const area
+      = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = Shape.circle;
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
+    this.color = color;
 
+    if (radius > 0) {
+      this.radius = radius;
```

Comment:
no need to do it here, you've already done it in the constructor
just throw an error if `radius <= 0`

Reply:
+